include/seqan3/range/container/constexpr_string.hpp

```diff
@@ -0,0 +1,430 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+#pragma once
+
+/*!\file
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ * \brief A constexpr string implementation to manipulate string literals at compile time.
+ */
+
+#include <type_traits>
+#include <array>
+
+namespace seqan3
+{
+
+/*!\brief Implements a constexpr string that can be used for compile time computations.
+ * \ingroup container
+ *
+ * This class provides a string type that can be constructed, evaluated and operated on at compile time.
+ * The stored string can be accessed as either a std::string or a c-style string through the respective member functions.
+ * All operations are constexpr except the conversion functions.
+ * \remark Internally the string stores an array of size `N+1` containing the `0`-byte to allow conversion to
+ * c-style strings. This also means that when creating an instance of this string from a string literal containing
+ * a `0`-byte the size of this instance is one less.
+ */
+template <std::size_t N>
+class constexpr_string
+{
+protected:
+    //!\brief Alias for the underlying data type.
+    using data_type = std::array<char, N + 1>;
+
+    //!\brief Gives access to the merge constructor to constexpr_string with different size.
+    template <std::size_t _N>
+    friend class constexpr_string;
+
+    //!\brief The internal string stored as array including \0-byte as last character.
+    data_type _lit;
+
+    /*!\brief Constructs new constexpr_string by merging two other constexpr_strings.
+     * \param lhs left-hand-side of constexpr_string to merge.
+     * \param rhs right-hand-side of constexpr_string to merge.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    template <std::size_t N1>
+    constexpr constexpr_string(constexpr_string<N1> const & lhs,
+                            constexpr_string<N - N1> const & rhs) noexcept
+    {
+        for (unsigned i = 0; i < N1; ++i)
+            _lit[i] = lhs[i];
+        for (unsigned i = N1; i < N + 1; ++i)
+            _lit[i] = rhs[i - N1];
+    }
+
+public:
+
+    /*!\name Public member types
```

Comment:
Must "Member types" maybe?

Reply:
Updated to STL.

Reply:
It still should be just "~~Public~~ Member types"